### ISSUE TITLE
[REFACTORING] Take away ImageLoader.getFromDisk() method from the main thread 

### ISSUE THREAD

**Issue by K1rakishou** [Adamantcheese/Kuroba#220]
As I have already said in #215 the new method (ImageLoader.getFromDisk()) calls the File API on the main thread which may (and it does) cause lags when disk is busy by the other threads/processes. It shouldn't be hard to do it becase it's already kinda asynchronous (it returns ImageContainer which can be used for the request cancelling) and what is left to do is to jsut move the code onto some executor. 

Damn, I have just noticed that ImageLoader is a part of the Volley library. I probably shouldn't have changed it at all.

**Comment by Adamantcheese** [comment]
Yeah I'm not super sure about pulling this one if you've modified the included Volley library and it's probably going to be changed to the upstream one like Floens did on Clover, at which point you wouldn't be able to change it. You'd still be able to override and use your own custom rolled class for it, but that shouldn't really be necessary.

**Comment by K1rakishou** [comment]
Well, I mean, those changes I made in Volley have already been merged. 
On the other hand, I reverted most of the volley changes in this (#240) PR and extracted them into ImageLoaderV2 (I couldn't come up with a better name) which is not a part of the Volley. But I had to extract some private interfaces from the ImageLoader (and one method) so I could access them in ImageLoaderV2. I also deprecated all calls to the old ImageLoader because we NEED to get rid of it completely.
